packages/amplify-cli/src/lib/migrate-project.js

```diff
@@ -0,0 +1,188 @@
+const fs = require('fs-extra');
+const path = require('path');
+const ora = require('ora');
+const os = require('os');
+
+const spinner = ora('');
+const { prompt } = require('gluegun/prompt');
+const { print } = require('gluegun/print');
+const {
+  getDotConfigDirPath,
+  getProjectConfigFilePath,
+  getAmplifyMetaFilePath,
+  getLocalEnvFilePath,
+  getProviderInfoFilePath,
+  getBackendConfigFilePath,
+  getGitIgnoreFilePath,
+} = require('../extensions/amplify-helpers/path-manager');
+
+async function migrateProject(plugins) {
+  try {
+    let projectConfigFilePath;
+    try {
+      projectConfigFilePath = getProjectConfigFilePath();
+    } catch (e) {
+      // New project, hence not able to find the amplify dir
+      return;
+    }
+
+    const projectConfig = JSON.parse(fs.readFileSync(projectConfigFilePath));
+    if (projectConfig.projectPath && await prompt.confirm('We detected the project was initialized using an older version of the CLI. Do you want to migrate the project, so that it is compatible with the latest version of the CLI?')) {
+      // This is an older project & migration is needed
+      generateNewProjectConfig(projectConfig, projectConfigFilePath);
+      generateLocalEnvInfo(projectConfig);
+      generateAwsLocalInfo();
+      generateTeamProviderInfo();
+      generateBackendConfig();
+      generateGitIgnoreFile();
+    }
+
+    // Give each category a chance to migrate their respective files
+
+    const categoryMigrationTasks = [];
+    const amplifyMetafilePath = getAmplifyMetaFilePath();
+    const amplifyMeta = JSON.parse(fs.readFileSync(amplifyMetafilePath));
+    const initializedCategories = Object.keys(amplifyMeta);
+
+    const categoryPlugins = {};
+    plugins.forEach((plugin) => {
```

Comment:
Don't we have a method for this?

Reply:
There's no helper function. The helper functions use context.

Reply:
Will change the logic for this.